File: src/utils/logger.ts
export interface Logger {
  debug(message: string, ...args: unknown[]): void;
  info(message: string, ...args: unknown[]): void;
  warn(message: string, ...args: unknown[]): void;
  error(message: string, ...args: unknown[]): void;
}

export function createLogger(debug: boolean = false): Logger {
  const prefix = "[context-graph]";
  return {
    debug: (message, ...args) => {
      if (debug) console.debug(`${prefix} ${message}`, ...args);
    },
    info: (message, ...args) => console.info(`${prefix} ${message}`, ...args),
    warn: (message, ...args) => console.warn(`${prefix} ${message}`, ...args),
    error: (message, ...args) =>
      console.error(`${prefix} ${message}`, ...args),
  };
}

File: python-sdk/src/graphmind_context_graphs/utils/logger.py
import logging

PREFIX = "[context-graph]"


def create_logger(debug: bool = False) -> logging.Logger:
    logger = logging.getLogger("graphmind_context_graphs")
    if not logger.handlers:
        handler = logging.StreamHandler()
        handler.setFormatter(logging.Formatter(f"{PREFIX} %(message)s"))
        logger.addHandler(handler)
    logger.setLevel(logging.DEBUG if debug else logging.INFO)
    return logger
